bundles/CoreBundle/src/Controller/PublicServicesController.php

```diff
@@ -49,6 +49,22 @@ public function thumbnailAction(Request $request): RedirectResponse|StreamedResp
             'file_extension' => $requestedFileExtension,
         ];
 
+        $asset = Asset::getById($config['asset_id']);
+        if($asset instanceof Asset\Image) {
+            // Return 404 if the requested thumbnail format is disabled from the config
+            $thumbnailFormats = ThumbnailConfig::getAutoFormats();
+            if(!in_array($requestedFileExtension, ['jpg', 'jpeg'])) {
+                if(empty($thumbnailFormats))
+                    return new Response("HTTP/1.1 404 Not Found\n Requested thumbnail format is disabled", 404);
+
+                foreach($thumbnailFormats as $autoFormat => $autoFormatConfig) {
+                    if(!$autoFormatConfig['enabled'] && $requestedFileExtension == $autoFormat) {
```

Comment:
```suggestion
                    if($requestedFileExtension == $autoFormat && !$autoFormatConfig['enabled']) {
```
imho by swapping order, is easier to read